### PR TITLE
[Extension] Added `Hide label` btn in user preferences

### PR DESCRIPTION
**Pull request by NikhilSharma03**
**Description**

- Added `Hide label` in user preferences

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/UserPreferences.js [hunk 1]
```diff
@@ -95,10 +108,17 @@ class UserPreference extends React.Component {
       startOnZoom : props.startOnZoom,
       tabVal : 0,
       userPrefs : ExtensionPointSchemaValidator("user_prefs")(),
-      providerType : ''
+      providerType : '',
+      showBtnLabels : true
     };
   }
 
+  showBtnLabelHandler = () => {
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply ui/components/UserPreferences.js:116]: @NikhilSharma03 what do you think?

File: ui/components/UserPreferences.js [hunk 2]
```diff
@@ -300,7 +324,38 @@ class UserPreference extends React.Component {
             <MesherySettingsPerformanceComponent />
           }
           {tabVal == 2 && userPrefs && providerType != 'local' &&
-            <ExtensionSandbox type="user_prefs" Extension={(url) => RemoteUserPref({ startOnZoom, handleToggle, url })} />
+            <div>
```
Comment (leecalcote) [inline ui/components/UserPreferences.js:327]: @utkarsh-pro this should be an empty `div`, yes? The contents of this extension point should come from the connected remote provider, if that remote provider offers this capability.

Comment (utkarsh-pro) [reply ui/components/UserPreferences.js:327]: @leecalcote, yes. @NikhilSharma03 I think the changes that are done here should rather be done on the provider side.

Comment (utkarsh-pro) [reply ui/components/UserPreferences.js:327]: I see that a table container is added, is there some issue that you faced while adding this to the extension @NikhilSharma03?

Comment (NikhilSharma03) [reply ui/components/UserPreferences.js:327]: > @leecalcote, yes. @NikhilSharma03 I think the changes that are done here should rather be done on the provider side.

Does provider side mean here :sweat_smile:  https://github.com/meshery/meshery/blob/a541cda0a171c470a75f78ef736f7427f5c6c8f8/ui/components/NavigatorExtension.js

**Comment by leecalcote** [comment]
@NikhilSharma03 please close when this has transitioned to the remote provider.

**Comment by NikhilSharma03** [comment]
@leecalcote I think I am a bit confused here. That is if we want to land that `Hide Label Button` here in user preferences 

Screenshot from 2021-11-04 20-03-58

Then we have to made those changes here in meshery repo where UserPreference.js file resides.
If we want all changes in remote provider then the `Hide Label Button` button should reside in MeshMap then.

**Comment by leecalcote** [comment]
@NikhilSharma03 be sure that you follow up with @Abhishek-kumar09

**Pull request closed without merging**